src/views/TaskEdit.vue

```diff
@@ -0,0 +1,179 @@
+<script setup lang="ts">
+import { ref, onMounted, nextTick } from 'vue';
+import { useRoute, useRouter } from 'vue-router';
+import TaskService from '@/services/TaskService';
+import DataTable from 'primevue/datatable';
+import Column from 'primevue/column';
+import Button from 'primevue/button';
+
+interface Task {
+  id: string;
+  title: string;
+  description: string;
+  status: string;
+  ownerId: string;
+  createdAt: string;
+  modifiedAt: string;
+  blockedBy: string;
+  duplicateOf: string;
+  relatedTo: string;
+}
+
+const route = useRoute();
+const router = useRouter();
+const taskService = new TaskService();
+const task = ref<Task | null>(null);
+const loading = ref(false);
+const statusOptions = ['PENDING', 'OPEN', 'IN_PROGRESS', 'CLOSED', 'REJECTED'];
+
+const startEditing = (row: any, field: string) => {
+  row.editing = field;
+  nextTick(() => {
+    const inputElement = document.querySelector('.editable-input') as HTMLElement | null;
+    inputElement?.focus();
+  });
+};
+
+const stopEditing = (row: any) => {
+  row.editing = null;
+};
+
+const fetchTask = async () => {
+  loading.value = true;
+  try {
+    const projectId = route.params.projectId as string;
+    const taskId = route.params.taskId as string;
+    task.value = await taskService.getTask(projectId, taskId);
+  } catch (error) {
+    console.error('Error fetching task:', error);
+  } finally {
+    loading.value = false;
+  }
+};
```

Comment:
You’re using loading as a ref to manage the task fetching state. You can make the logic cleaner by using Vue's lifecycle hooks to set the loading state accordingly.

example:  onMounted